app/models/bus_stops_route.rb

```diff
@@ -9,4 +9,45 @@ class BusStopsRoute < ApplicationRecord
   validates :direction, presence: true
   validates_uniqueness_of :sequence, scope: %i[route direction]
   validates_uniqueness_of :bus_stop, scope: %i[route direction]
+
+  def self.import(stop_hash)
+    stop_hash.each_pair do |route_dir, direction_variants|
+      variants = direction_variants.values
+      longest_variant = variants.max_by(&:length)
+      other_variants = variants - longest_variant
+      other_variants.each do |other_variant|
+        other_variant.each.with_index do |stop, sequence|
+          unless longest_variant.include? stop
+            # if sequence is zero, then it is the first stop. We can't rely on
+            # its previous stop like we do in the corresponding else statement.
+            if sequence == 0
```

Comment:
In case you would prefer, you could also write `.zero?`

But perhaps in this case `if stop == other_variant.first` might be the *most* communicative?

Reply:
I think I'll go with `if stop == other_variant.first`.